functions/nvm.fish

```diff
@@ -184,8 +184,8 @@ end
 
 function _nvm_find_up --argument-names path file
     test -e "$path/$file" && echo $path/$file || begin
-        test "$path" != / || return
-        _nvm_find_up (command dirname $path) $file
+        test ! -z "$path" || return
+        _nvm_find_up (string replace --regex -- '/[^/]*$' "" $path) $file
```

Comment:
Might be worth adding a comment here, so one isn’t tempted to refactor to the simpler `dirname` approach in the future.

Reply:
Another thought: Is a `while` loop faster than recursion? Don’t know what the overhead of calling functions in fish is.

Reply:
Well, it doesn't seem to make much difference when using the following version:

```fish
function _nvm_find_up --argument-names path file
    while true
        test -e "$path/$file" && echo $path/$file && return
        set path (string replace --regex -- '/[^/]*$' "" $path)
        test "$path" = "" && return 1
    end
end
```

I even got mixed results when I ran it more than once. Sometimes, this version is slower by 2~3 ms. 

Keep in mind I'm 40-ish directories deep in the directory tree. When you're in the same directory, even the original version using `dirname` runs at around 200 microseconds. 🤓

Reply:
> Might be worth adding a comment here, so one isn’t tempted to refactor to the simpler dirname approach in the future.

Ah, comments, yeah, let me come up with a good excuse to avoid having to write one. 

Got it!

It's fishier (as in more idiomatic in Fish) to use builtins, as they're usually faster and consistent across OSes, therefore one shouldn't be tempted to switch to an external command (which may not even be available in the host system).